programs/github

```diff
@@ -21,13 +21,48 @@ function progressbar(f)
 	return left .. right
 end
 
+-- get flag arguments
+local acceptedFlags = {
+	'-b' = 'branch',
+	'--branch' = 'branch',
+	'-t' = 'tag',
+	'--tag' = 'tag',
+	'-a' = 'auth',
+	'--auth' = 'auth'
+}
+
+local findFlag = function(arg) return acceptedFlags[arg] end
+
+local passed = {...}
+local args = {}
+local flags = {}
+local flagValue = ''
+for k, arg in pairs(passed) do
+	local flagName = findFlag(arg)
+	if flagName and arg ~= flagValue then
+		flags[flagName] = passed[k + 1]
+		flagValue = flags[flagName]
+	elseif arg ~= action and arg ~= flagValue then
+		table.insert(args, arg)
+	end
+end
+
+if flags.branch and flags.tag then
+	return printError('please specify a branch or tag.')
```

Comment:
Yeah, this conveys the wrong message. "--branch and --tag cannot both be specified" would be more accurate.

Note that `git clone` doesn't actually "support" tags, but `git clone -b a_tag` tends to do the right thing anyway